modules/distributor/receiver/shim.go

```diff
@@ -279,6 +291,7 @@ func (r *receiversShim) ConsumeTraces(ctx context.Context, td ptrace.Traces) err
 // ReportFatalError implements component.Host
 func (r *receiversShim) ReportFatalError(err error) {
 	_ = level.Error(log.Logger).Log("msg", "fatal error reported", "err", err)
+	r.StopAsync()
```

Comment:
I think you need to pass the err to `fatal` channel.

Reply:
Mea Culpa
I added the err to fatal channel logic to our fork which is on an older version and completely failed at updating the shim here correctly